datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ingest/source/UpsertIngestionSourceResolver.java

```diff
@@ -46,55 +50,51 @@ public UpsertIngestionSourceResolver(final EntityClient entityClient) {
   public CompletableFuture<String> get(final DataFetchingEnvironment environment) throws Exception {
     final QueryContext context = environment.getContext();
 
-    return GraphQLConcurrencyUtils.supplyAsync(
-        () -> {
-          if (IngestionAuthUtils.canManageIngestion(context)) {
-
-            final Optional<String> ingestionSourceUrn =
-                Optional.ofNullable(environment.getArgument("urn"));
-            final UpdateIngestionSourceInput input =
-                bindArgument(environment.getArgument("input"), UpdateIngestionSourceInput.class);
+    if (!IngestionAuthUtils.canManageIngestion(context)) {
```

Comment:
Moved everything but ingestProposal out of the async portion... rest should be really fast and makes testing simpler